modules/core/include/opencv2/core/hal/intrin_cpp.hpp

```diff
@@ -574,6 +574,49 @@ OPENCV_HAL_IMPL_REDUCE_MINMAX_FUNC(v_reduce_min, std::min)
 For 32-bit integer and 32-bit floating point types. */
 OPENCV_HAL_IMPL_REDUCE_MINMAX_FUNC(v_reduce_max, std::max)
 
+/** @brief Count the 1 bits in the vector and return 4 values
+
+Scheme:
+@code
+{A1 A2 A3 ...} => popcount(A1)
+@endcode
+Any types but result will be in v_uint32x4*/
+template<typename _Tp, int n> inline v_uint32x4 v_popcount(const v_reg<_Tp, n>& a)
+{
+    static unsigned char popCountTable[] =
```

Comment:
Thank you for your review and your time.

To answer your question about 

> What is about the performance of "int" type vs "unsigned char"?,

 I assume it means
"how much will the performance change if reducing the number of addition, and return the vector in ```v_uint8x16``` rather than ```v_uint32x4``` ?"
I tried writing small experiment code, and indeed, reducing the number of addition seems promising.
I agree to your suggestion, but it will take time to commit due to following two reasons.
 - the result will easily overflow
 - there are no ```v_reduce``` function for 16 elements vector
For the first point, I need to modify ```stat.cpp```.  It shouldn't take so time.
For the second point, I need to add new function, so it will take time.

let me take couple of more days to modify ```v_popcount``` as you suggest.
I totally agree to your suggestion.

Best regards.